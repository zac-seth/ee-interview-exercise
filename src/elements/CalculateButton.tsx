import { ReactElement } from 'react';

import { Button } from './Button.style';
import { ValuelessButtonProps } from './shared';

export const CalculateButton = ({
  label,
  onClick
}: ValuelessButtonProps): ReactElement => (
  <Button onClick={onClick} isCalculate>
    {label}
  </Button>
);
