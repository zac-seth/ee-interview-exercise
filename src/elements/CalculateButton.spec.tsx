import { fireEvent, render, screen } from '@testing-library/react';

import { CalculateButton } from './CalculateButton';

describe(`CalculateButton`, () => {
  it(`Should render a calculate button`, () => {
    const handleClick = () => {};
    render(<CalculateButton label='+' onClick={handleClick} />);

    expect(screen.getByText(`+`)).toMatchSnapshot();
  });

  it(`Should invoke the handler when clicked`, () => {
    const handleClick = jest.fn();
    render(<CalculateButton label='+' onClick={handleClick} />);

    fireEvent.click(screen.getByText(`+`));

    expect(handleClick).toHaveBeenCalledTimes(1);
  })
});
