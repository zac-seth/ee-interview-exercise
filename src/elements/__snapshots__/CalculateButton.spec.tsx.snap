// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculateButton Should render a calculate button 1`] = `
.c0 {
  font-family: 'Lato',sans-serif;
  font-weight: 400;
  color: #666;
  text-shadow: 0 0 5px #1A94D2,0 0 15px #228DFF;
  font-size: 4rem;
  line-height: 4rem;
  text-align: center;
  padding: 1rem;
  margin: 0;
  width: 8rem;
  height: 8rem;
  background-color: inherit;
  color: #1a94d2;
  border: 0;
  cursor: pointer;
  background-color: #1a2b1a;
}

.c0:hover {
  background-color: #1a1a1a;
}

.c0:hover {
  background-color: #1d2e1d;
}

<button
  class="c0"
>
  +
</button>
`;
